### PR TITLE
docs: add standards for UI, form fields and regex

### PR DESCRIPTION
**Pull request by rajkumar-osmosys**
## Summary by CodeRabbit

- **Documentation**
	- Updated UI standards documentation to include guidelines for data testing, date/time display, form design, grid/table guidelines, tooltips, responsiveness, and more.

### CODE REVIEW

File: coding-standards/UI.md [hunk 1]
```diff
@@ -0,0 +1,505 @@
...
+| 1   | Alignment                         | Design                   | While creating components in an app, make sure they align well from a UI perspective.                                                                                                                                                                                                                                                                                 |
+| 2   | Bulk import                       | Import process           | Bulk import (users or anything) should be done asynchronously in the background.. we cannot hold the UI / portal / user waiting for a long time while import is happening.                                                                                                                                                                                            |
```
[the remark this answers is not in the document]
Comment (rajkumar-osmosys) [reply coding-standards/UI.md]: remove extra dot.

File: coding-standards/UI.md [hunk 2]
```diff
@@ -0,0 +1,505 @@
...
+
+[![Picture4.jpg](https://soochna.osmosys.co/uploads/images/gallery/2016-11-Nov/scaled-840-0/Picture4.jpg)](https://soochna.osmosys.co/uploads/images/gallery/2016-11-Nov/Picture4.jpg)
+
+The above image shows three different column widths for the same data type (money) – such  things  shall  be  avoided,  unless explicitly asked by the client.
+
+## Properties
+
+This document describes the various type of fields and their properties.
+
+<table border="0" cellspacing="0" id="bkmrk-use-db-datatype-db-l">
```
Comment (osm-vishnukyatannawar) [inline coding-standards/UI.md:157]: Convert this to markdown table instead. Like the above UI standards.

File: coding-standards/UI.md [hunk 3]
```diff
@@ -0,0 +1,505 @@
...
+
+Fields are aligned horizontally and vertically.  In addition to alignment, the spacing between fields both horizontally and vertically is important. 
+
+**Label Vertical alignment** – all labels must be either top aligned or center aligned. Osmosys by default uses top alignment.
+
+**Label Horizontal alignment** – all labels must be either left aligned or right aligned. Osmosys by default uses left alignment.
+
+**Label / Field Vertical spacing** - This is the space between two fields in the same column. It is very important to maintain this same across all fields in the page & application.
+
+<table style="width: 562.5px; height: 157px;">
```
Comment (osm-vishnukyatannawar) [inline coding-standards/UI.md:109]: Kindly convert it to markdown table

File: coding-standards/UI.md [hunk 4]
```diff
@@ -0,0 +1,505 @@
...
+      <tr style="height: 31px;">
+         <td style="width: 295px; height: 31px;">Label Horizontal Alignment
+         </td>
+         <td style="width: 265.5px; height: 31px;">Left
+         </td>
+      </tr>
+      <tr style="height: 31px;">
+         <td style="width: 295px; height: 31px;">Label Font Type
+         </td>
+         <td style="width: 265.5px; height: 31px;"></td>
```
Comment (osm-vishnukyatannawar) [inline coding-standards/UI.md:131]: These label font type and label font size there are no values against them.

File: coding-standards/UI.md [hunk 5]
```diff
@@ -0,0 +1,505 @@
...
+
+ 
+## Sizes
+
+Proper field sizes are essential for comfortable data entry and good appearance of the user interface. Following points are to be kept in serious consideration while designing user interfaces:
+
+1. **Relevance:** Proper size for each field is important for a good user experience. An email with 50 characters shall have the field width to accommodate that size. Similarly, the field for Age can have a size that takes 3 characters. If we are  taking notes or comments, then that fields should reflect 
+2. **Uniformity:** All fields of the same type in a form / application shall have the same size. This is properly achieved when we follow a well designed CSS. The same proper sizing shall be followed even in grids.
+
+[![Picture4.jpg](https://soochna.osmosys.co/uploads/images/gallery/2016-11-Nov/scaled-840-0/Picture4.jpg)](https://soochna.osmosys.co/uploads/images/gallery/2016-11-Nov/Picture4.jpg)
```
Comment (osm-vishnukyatannawar) [inline coding-standards/UI.md:149]: Cannot keep links which redirect to soochna. THis is open source and the links are going back to soochna which is behind a login.

Create a assets folder and keep them inside that.
coding-standards - That folder can stay here in this folder.

Use the image from there.

File: coding-standards/UI.md [hunk 6]
```diff
@@ -0,0 +1,505 @@
...
+         <td align="left" style="height: 49px;"> </td>
+         <td align="left" style="height: 49px;"> </td>
+         <td align="left" style="height: 49px;"> </td>
+         <td align="left" style="height: 49px;"> </td>
+         <td align="left" style="height: 49px;">Must be in <span style="color: #008000;">GREEN</span> color.</td>
+      </tr>
+   </tbody>
+</table>
+
+# Standard regular expressions to be used for form validations across the technologies
```
Comment (osm-vishnukyatannawar) [inline coding-standards/UI.md:433]: If we want to keep this here then delete the regex file which we created before: 

https://github.com/rajkumar-osmosys/dev-standards/blob/patch-1/coding-standards/RegEx.md

**Review by osm-vishnukyatannawar — CHANGES_REQUESTED** [review]

File: coding-standards/UI.md (hunk 2 above)
Comment (LakshayaT) [reply coding-standards/UI.md]: Resolved in https://github.com/OsmosysSoftware/dev-standards/pull/56/commits/f0790d0686275585fd1798073a972c3e86ce88b1

File: coding-standards/UI.md (hunk 3 above)
Comment (LakshayaT) [reply coding-standards/UI.md]: Resolved in https://github.com/OsmosysSoftware/dev-standards/pull/56/commits/f0790d0686275585fd1798073a972c3e86ce88b1

File: coding-standards/UI.md (hunk 4 above)
Comment (LakshayaT) [reply coding-standards/UI.md]: Unsure of the values to keep, the table was converted to markdown, but kept these values blank only for now

File: coding-standards/UI.md (hunk 5 above)
Comment (LakshayaT) [reply coding-standards/UI.md]: Resolved in https://github.com/OsmosysSoftware/dev-standards/pull/56/commits/f0790d0686275585fd1798073a972c3e86ce88b1

**Review by osm-vishnukyatannawar — APPROVED** [review]

**Pull request merged**